### ISSUE TITLE
too many items

### ISSUE THREAD

**Issue by HybridDog** [minetest-streets/streets#85]
l noticed that streets (the new version) adds an enormous amount of items:

``````
1516  streets
494   technic
416   default
326   stairs
268   pipeworks
162   mesecons
131   diplazer
124   xconnected
108   moretrees
102   take
100   wrench
91    beer_test
89    nether
89    scifi_nodes
83    moreblocks
81    digilines
80    xpanes
69    foodblock
68    crops
59    hydro
55    extrablocks
49    farming
46    xtraores
43    bobblocks
42    factory
42    potions
41    cannons
41    farming_plus
40    riesenpilz
40    my_future_doors
39    darkage
39    my_castle_doors
36    my_sliding_doors
36    magic_wand
36    bushes
35    shooter
33    moreladders
33    morefences
31    snow
30    multitest
30    my_hidden_doors
30    heads
29    computer
28    sumpf
27    doors
25    moreores
25    moreplants
24    stairsshine
24    my_fancy_doors
24    realclocks
22    flowers
22    painting
20    ac
19    usesdirt
19    laser
19    hemp
19    my_misc_doors
19    pathv6alt
18    mesecons_pistons
17    workers
17    jdukebox
17    mesecons_luacontroller
17    jdc_lamps
16    mesecons_microcontroller
16    mo
15    wool
15    seaplants
15    mesecons_extrawires
15    dye
15    my_default_doors
14    urban
14    caca
14    dogblocks
14    fireworks
14    special_picks
14    titanium
14    mesecons_gates
14    my_saloon_doors
12    my_old_doors
12    seaglass
12    fancy_elevator
12    mesecons_lightstone
11    projects
11    another_charcoal
10    lantern
10    firearms
9     nyanland
8     mesecons_delayer
8     diamonds
8     morecobblenodes
8     steel
8     trimet
8     gt1111
7     seacobble
7     weirdores
7     my_door_wood
7     seastonebrick
7     seagravel
7     seastone
6     greenscreens
6     mining_plus
6     my_cottage_doors
6     extinguisher
6     cutting_board
6     travelnet
6     myminegate
6     quartz
6     compressed_cobble
6     vines
6     moretechnictools
6     clams
5     conifers
5     nuke
5     vessels
5     more_chests
4     beds
4     mesecons_pressureplates
4     caches
4     mesecons_detector
4     digipad
4     ufos
4     music
4     bucket
4     my_garage_door
4     seawrecks
4     jungletree
4     money
4     tnt
4     connected_chests
4     pathogen
4     mesecons_movestones
3     unified_inventory
3     moremesecons_dual_delayer
3     walls
3     lavacooling
3     poisonivy
3     home_builder
3     mesecons_materials
3     fire
3     mudbrick
3     mesecons_random
3     moremesecons_wireless
3     eridium
2     moremesecons_jammer
2     noairblocks
2     handle_schematics
2     mesecons_blinkyplant
2     mesecons_commandblock
2     moremesecons_injector_controller
2     mesecons_lamp
2     mesecons_solarpanel
2     wz2100
2     moremesecons_temporarygate
2     teleporters
2     conveyor_belt
2     tetris
2     moremesecons_conductor_signalchanger
2     mesecons_hydroturbine
2     jukebox
2     replacer
2     aclean
2     moremesecons_adjustable_player_detector
2     moremesecons_entity_detector
2     mesecons_walllever
2     trash_can
2     mesecons_switch
2     mesecons_receiver
2     homedecor
2     glow
2     rotten_papyrus
2     nyancat
2     mesecons_torch
2     futon
2     worldedit
2     moremesecons_adjustable_blinkyplant
2     plasmalamp
2     moremesecons_commandblock
2     simple_robots
2     mesecons_insulated
2     moremesecons_switchtorch
2     mese_fence
2     itemframes
2     moremesecons_signalchanger
2     mesecons_button
2     sealamps
2     slidebail
1     regelblock
1     moremesecons_playerkiller
1     digilines_lcd
1     maze
1     ?
1     boats
1     simple_machines
1     trashcan
1     mapp
1     tmt
1     moremesecons_teleporter
1     mesecons_stickyblocks
1     rnd_trees
1     far_chest_accessor
1     colour_carrier
1     screwdriver
1     air
1     unknown
1     auto_tree_tap
1     sun_flowers
1     creative
1     metatools
1     moremesecons_luacontroller_tool
1     moss
1     digiline_touchscreen
1     digilines_inventory
1     ruins
1     shears
1     digilines_rtc
1     mesecons_noteblock
1     linemaker
1     nodebox_creator
1     command_tool
1     mesecons_teleporter
1     bones
1     tbm
1     moremesecons_sayer
1     wield3d
1     mesecons_powerplant
1     digilines_lightsensor
1     ltool
1     radio
1     moremesecons_igniter
1     gbpinvisiable
1     locked_sign```
``````

**Comment by webdesigner97** [comment]
Yep... This amount became huge... That's because ebery single marking on asphalt has all shapes of the circular_saw registered, too... IU'm not happy with this because it makes the game load forever... We need to discuss this...

**Comment by HybridDog** [comment]
indeed, l just rebased my streets mod to the new version and now there're 1516  items

**Comment by webdesigner97** [comment]
We really need to remove some stuff, but I have to wait for the others... They need to decide as well...

**Comment by cheapie** [comment]
I made a forum thread a while back mentioning this and (once again) calling
for texture override support. It died out yet again, and yet we pretty much
need this in order to bring this number down.

**Comment by Thomas--S** [comment]
I think that once there will be meta-based node-def. It is only a question of time. Until then, we have to live with the huge amount of nodes. I am testing this mod on my server, and surprisingly it isn't as slow as you would expect. So I think the many nodes are acceptable as long as there is no meta-set node-def. As soon as this feature arrives, we can start to develop Streets 3.0

**Comment by HybridDog** [comment]
Maybe in some cases connected nodebox drawtype can be used.

**Comment by Thomas--S** [comment]
The only place where they _could_ be used, are the bigpoles. But we all agreed that not using connected nodeboxes  there is better. Everywhere else, we use them already, if it makes sense.

**Comment by Thomas--S** [comment]
I'll remove this from the Streets 2.0 Milestone, as it is unlikely, that the needed meta-based node-def feature will come soon. I think that this problem here should be solved in Streets 3.0 (as soon as Minetest has meta-based node-def)

**Comment by Thomas--S** [comment]
(Partially) fixed by dd62ee45a3cd5e301c497b5a3f93cf6b4d644950

**Comment by Thomas--S** [comment]
:arrow_right: #101
